### PR TITLE
Syndicate Encryption Key buff

### PR DESCRIPTION
**Pull request by GoodWheatley**
## About the PR

This PR allows the syndicate encryption key to access all radio channels. While I've kept the price at 2 TC, I'm willing to raise it if I get feedback to do that. 

Now I know some of you might already be typing out a response saying that this doesn't make any sense, so let me explain. Right now communicating with other traitors is nigh impossible to do. Unless you want to hang out in the bar saying codewords for half the shift, or if you have an objective that reveals another traitor's identity, you're on your own. The addition of the encryption key helped alleviate this a little since you don't have to wear a headset that makes it absurdly obvious you're a bad guy, but it still takes two totally independent traitors to buy this to make it worth it. I think players still need a little more incentive to buy this, hence adding all the radio channels. More traitors buying this would make it much easier for traitors to discover each other and cooperate (or kill each other), encouraging social gameplay. And since you actually have to spend 2 TC up front to get this, you'll have to get 2 extra traitors together to buy that super surplus. I also removed the headset from the uplink, since with the addition of encryption keys it doesn't need to be there anymore.

**Media**

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl:
- add: A recently discovered security vulnerability in Nanotrasen radio frequencies has been exploited by the Syndicate, allowing the Syndicate Encryption Key to access all station radio channels.

### CODE REVIEW

File: Resources/Prototypes/Catalog/uplink_catalog.yml [hunk 1]
```diff
@@ -268,16 +268,15 @@
...
-- type: listing
-  id: UplinkHeadset
-  name: uplink-headset-name
-  description: uplink-headset-desc
-  productEntity: ClothingHeadsetAltSyndicate
-  cost:
-    Telecrystal: 4
-  categories:
-  - UplinkUtility
+# made irrelevant by encryption keys
```
Comment (EmoGarbage404) [inline Resources/Prototypes/Catalog/uplink_catalog.yml:271]: remove unused code

File: Resources/Locale/en-US/store/uplink-catalog.ftl [hunk 2]
```diff
@@ -193,7 +193,7 @@ uplink-syndicate-segway-crate-desc = Be an enemy of the corporation, in style!
 uplink-revolver-cap-gun-name = Cap Gun
 uplink-revolver-cap-gun-desc = Looks almost like the real thing! Ages 8 and up.
 
-uplink-syndicate-stamp-name = Syndicate Rubber Btamp
+uplink-syndicate-stamp-name = Syndicate Rubber Stamp # In loving memory, Btamp, ???-2/1/23
```
Comment (EmoGarbage404) [inline Resources/Locale/en-US/store/uplink-catalog.ftl:196]: what

**Review by EmoGarbage404 — CHANGES_REQUESTED** [review]

File: Resources/Locale/en-US/store/uplink-catalog.ftl (hunk 2 above)
Comment (EmoGarbage404) [reply Resources/Locale/en-US/store/uplink-catalog.ftl:196]: oh wait btamp meme nvm

Comment (GoodWheatley) [reply Resources/Locale/en-US/store/uplink-catalog.ftl:196]: spotted a typo there when I was making the ftl entry

File: Resources/Prototypes/Catalog/uplink_catalog.yml (hunk 1 above)
Comment (GoodWheatley) [reply Resources/Prototypes/Catalog/uplink_catalog.yml]: done

**Review by metalgearsloth — DISMISSED** [review]
This gives nukies access to every channel.

File: Resources/Locale/en-US/store/uplink-catalog.ftl (hunk 2 above)
Comment (metalgearsloth) [reply Resources/Locale/en-US/store/uplink-catalog.ftl:196]: at least format the date correctly.

**Comment by GoodWheatley** [comment]
> This gives nukies access to every channel.

I don't think that's really a bad thing, the worst they can do with that is taunt the crew & know if sec's onto them during stealthops

**Comment by metalgearsloth** [comment]
https://github.com/space-wizards/space-station-14/issues/13782

They're not supposed to have access to it.

**Comment by GoodWheatley** [comment]
> > This gives nukies access to every channel.
> 
> Intentional, since I don't think that's really a bad thing, the worst they can do with that is taunt the crew & know if sec's onto them during stealth ops

**Comment by metalgearsloth** [comment]
I don't think you're familiar with when we deliberately gave them access to it and it went over poorly.

**Comment by GoodWheatley** [comment]
> #13782
> 
> They're not supposed to have access to it.

But can't the pAI's that spawn in the nukie base use it?

**Comment by GoodWheatley** [comment]
Not sure why tests are failing, error messages don't seem related to the files I changed

**Comment by AlexMorgan3817** [comment]
Still very strange, this kills trick with hacking in caps' to get his radio key. I think simple syndie key should  be separated from "ALLCHANNELS"-key.
Or better add headset with unextractable ALLCHANNEL-key

**Comment by GoodWheatley** [comment]
> Still very strange, this kills trick with hacking in caps' to get his radio key. I think simple syndie key should be separated from "ALLCHANNELS"-key. 
As I said in the description, it's to encourage traitors to buy the key so there's more social gameplay among them. Splitting up the two different versions of the key just adds more uplink bloat for no good reason. Hacking into the captain's bedroom or stealing a command headset will give you all station channels, but it won't give you access to the traitor channel that is the key's main purpose.
> Or better add headset with unextractable ALLCHANNEL-key
That's exactly what I'm trying to avoid since then everyone knows you're a traitor.

**Comment by AlexMorgan3817** [comment]
> > Still very strange, this kills trick with hacking in caps' to get his radio key. I think simple syndie key should be separated from "ALLCHANNELS"-key. 
> 
> As I said in the description, it's to encourage traitors to buy the key so there's more social gameplay among them. Splitting up the two different versions of the key just adds more uplink bloat for no good reason. Hacking into the captain's bedroom or stealing a command headset will give you all station channels, but it won't give you access to the traitor channel that is the key's main purpose.
> 
> > Or better add headset with unextractable ALLCHANNEL-key
> 
> That's exactly what I'm trying to avoid since then everyone knows you're a traitor.
> 

_Need a traitor channel? Buy it. Problems, gamers?_

Okay, i understood what do you want and think it is nice, but 2 tc seems... too few, i guess

**Comment by Emisse** [comment]
#14110 is a more interesting solution to buffing traitor headset because it gives heads lockers boxes of keys to steal, and also doesnt just give them all access radio for doing nothing. I feel like this PR is just bypassing the encryption key system entirely when we can have interesting gameplay revolving around stealing keys for the headset. The only buff I feel like syndi headset needs is chameleon to prevent meta gaming.

**Comment by AlexMorgan3817** [comment]
> I feel like this PR is just bypassing the encryption key system entirely when we can have interesting gameplay revolving around stealing keys for the headset.

Yeah, bro, you are right, this what i somewhy forget.

> The only buff I feel like syndi headset needs is chameleon to prevent meta gaming.

Because syndie headset have 4 key slots and have syndie key in it, it has to become better with chameleon

**Comment by GoodWheatley** [comment]
> #14110 is a more interesting solution to buffing traitor headset because it gives heads lockers boxes of keys to steal, and also doesnt just give them all access radio for doing nothing. 

What I'm trying to encourage with this change is more traitor interaction, and while I do think #14110 is a good change, it's just going to lead to even more head office robberies which doesn't usually lead to any interactions besides getting thrown in brig. During our discussion on Discord, you said that you've observed multiple traitors buying the key and interacting with each other during rounds. That's the type of gameplay I'm trying to encourage, and I don't think this change will cause encryption key theft to no longer be a good strategy.

> doing nothing.

You're trading 2 TC and the ability for anyone who examines your headset to know you're valid for those radio channels. I think that's a balanced trade.

> I feel like this PR is just bypassing the encryption key system entirely when we can have interesting gameplay revolving around stealing keys for the headset. 

I don't really understand this argument, since the point of most traitor items is they're a way to quickly gain power, making you lose the ability to deny you're a traitor in exchange. Stealing keys for the headset either leads to emagging into multiple head lockers, which makes the emag even more of a must-buy, or just killing heads and taking their headsets, which leads to more boring hypopen gameplay. Giving traitors the option to skip this and get access to all the channels doesn't immediately make this gameplay irrelevant, since traitors could choose not to buy it over other items that might be more useful, and then later break into a sec outpost for that encryption key if they think they need it. If traitors don't frequently steal encryption keys, it's probably because they just don't need them right now. I'm not forcing them to buy this, merely making it a more attractive option to buy, because I think traitor rounds become much more fun when you can easily communicate with other traitors.

> The only buff I feel like syndi headset needs is chameleon to prevent meta gaming.

Even if it gets the chameleon ability (which can still be metagamed by interacting with the headset), it's still stuck between a item that costs 2 less TC and does exactly what the headset does when its first purchased, and an item that costs 2 more TC that gives you a full set of chameleon clothes. It just doesn't make sense for the headset to be in the uplink anymore, and no amount of encryption key slots will change that.

**Comment by Emisse** [comment]
why is traitors doing traitorous things bad. also traitors already have code words i think giving more incentive to bypass that with this pr is just ditching 2 game systems for an easy out

**Comment by Just-a-Unity-Dev** [comment]
how about jolhyomh lryeords

**Comment by Just-a-Unity-Dev** [comment]
codewords

**Comment by PixelTheKermit** [comment]
Having them access all channels doesn't seem that good tbh (then again I haven't played traitor in god knows how long so I'm probably not allowed to have an opinion on this)

**Comment by GoodWheatley** [comment]
@Emisse
@AlexMorgan3817 

I feel like the point of this change has gotten lost in all the arguing, so I’m just going to restate it so it's clear. **I am not trying to supersede encryption key theft with this change.** It’s something I’ve had on my todo list for months now, before the encryption key PR ever existed. The only thing that held me back from doing it for the syndie headset was the fact that you couldn’t wear it safely (I forgot about chameleon). The goal behind this change is to get people to buy the encryption key more, because unlike any other uplink item, it doesn’t have a guaranteed usefulness. As I said above, it takes two traitors to even make this item **useable**, and if nobody else bought it you now have a very fancy and illegal paperweight. The fact that you could potentially waste 2 TC on this and get nothing in return is completely out of line with how uplink items are designed. You need several players to independently buy this to make it useful, but nobody buys it, so it's worthless. My idea for fixing this was to add a guaranteed benefit of buying the key, hence the extra channels being added. The only reason I chose that specific way of buffing the key was because that’s how it works in 13, and I didn’t really see any issues with letting traitors access those channels. It’s just a means of gaining extra information, which is always useful to traitors who are actually trying to be stealthy instead of buying an esword + c4 bundle and going loud. While I do recognize this could reduce the rates of encryption key theft, it won’t eliminate it entirely since not every antag is going to be a traitor, and not every traitor is going to want to always buy this. If somebody can come up with a better static benefit for the key, I will gladly close this PR in favor of theirs. I want this item to be bought more, because I love the idea of traitors linking up and causing conflict between themselves, and creating a story that takes place entirely beyond the crew’s eyes. While I think this is the best way to get people to buy the key, I’ll be glad to see someone prove me wrong.

Quick note on the suggestion to give the headset chameleon:
That still doesn’t solve the issue of the key having a good chance to be worthless. Sure you won’t be immediately called out as a tator tot, but if nobody else has the traitor channel, you just spent 4 TC on a fancy headset with no common channel access. And if the syndie key can be covertly inserted into your headset, why even have the syndie headset in the uplink at this point? It’s just bloat that won’t ever be bought when you can just buy the key for half the price, which is what traitors are going to want over the extra key slots.

**Comment by PixelTheKermit** [comment]
> I feel like the...

I'm not reading all that
I'm happy for you though
Or sorry that happened

**Comment by mirrorcult** [comment]
> 
> I'm not reading all that I'm happy for you though Or sorry that happened

dont be a dumbass

**Comment by PixelTheKermit** [comment]
> > I'm not reading all that I'm happy for you though Or sorry that happened
> 
> dont be a dumbass

Sorry

**Comment by GoodWheatley** [comment]
> I'm not reading all that

tldr: The encryption key needs a guaranteed benefit to encourage traitors to buy it, since right now its usefulness is completely dependent on whether other traitors independently decided to buy it. Adding the extra channels is just a way to add that guaranteed benefit, not to bypass key theft and codewords.

**Comment by PixelTheKermit** [comment]
> > I'm not reading all that
> 
> tldr: The encryption key needs a guaranteed benefit to encourage traitors to buy it, since right now its usefulness is completely dependent on whether other traitors independently decided to buy it. Adding the extra channels is just a way to add that guaranteed benefit, not to bypass key theft and codewords.

Hm, I think T encryption keys shouldn't be able to access *all* channels, however maybe most aside from command and maybe sec?

**Comment by GoodWheatley** [comment]
> > > I'm not reading all that
> > 
> > 
> > tldr: The encryption key needs a guaranteed benefit to encourage traitors to buy it, since right now its usefulness is completely dependent on whether other traitors independently decided to buy it. Adding the extra channels is just a way to add that guaranteed benefit, not to bypass key theft and codewords.
> 
> Hm, I think T encryption keys shouldn't be able to access _all_ channels, however maybe most aside from command and maybe sec?

Removing the command channel from the key could work, but the sec channel is probably going to be a pretty significant factor in whether or not to buy it, since knowing if sec is onto you is extremely valuable information.

**Comment by Slava0135** [comment]
> > > > I'm not reading all that
> > > 
> > > 
> > > tldr: The encryption key needs a guaranteed benefit to encourage traitors to buy it, since right now its usefulness is completely dependent on whether other traitors independently decided to buy it. Adding the extra channels is just a way to add that guaranteed benefit, not to bypass key theft and codewords.
> > 
> > 
> > Hm, I think T encryption keys shouldn't be able to access _all_ channels, however maybe most aside from command and maybe sec?
> 
> Removing the command channel from the key could work, but the sec channel is probably going to be a pretty significant factor in whether or not to buy it, since knowing if sec is onto you is extremely valuable information.

Then just blast in security checkpoint or HoP office or even kill secoff.

**Comment by GoodWheatley** [comment]
> > > > > I'm not reading all that
> > > > 
> > > > 
> > > > tldr: The encryption key needs a guaranteed benefit to encourage traitors to buy it, since right now its usefulness is completely dependent on whether other traitors independently decided to buy it. Adding the extra channels is just a way to add that guaranteed benefit, not to bypass key theft and codewords.
> > > 
> > > 
> > > Hm, I think T encryption keys shouldn't be able to access _all_ channels, however maybe most aside from command and maybe sec?
> > 
> > 
> > Removing the command channel from the key could work, but the sec channel is probably going to be a pretty significant factor in whether or not to buy it, since knowing if sec is onto you is extremely valuable information.
> 
> Then just blast in security checkpoint or HoP office or even kill secoff.

The point of all the extra channels is to get people to buy the key and then use the traitor channel by extent, if you just want the sec channel you can do any of those things

**Comment by PixelTheKermit** [comment]
> > > tldr: The encryption key needs a guaranteed benefit to encourage traitors to buy it, since right now its usefulness is completely dependent on whether other traitors independently decided to buy it. Adding the extra channels is just a way to add that guaranteed benefit, not to bypass key theft and codewords.
> > 
> > 
> > Hm, I think T encryption keys shouldn't be able to access _all_ channels, however maybe most aside from command and maybe sec?
> 
> Removing the command channel from the key could work, but the sec channel is probably going to be a pretty significant factor in whether or not to buy it, since knowing if sec is onto you is extremely valuable information.

I guess that's true, but that's still very useful, so 

> > > > > I'm not reading all that
> > > > 
> > > > 
> > > > tldr: The encryption key needs a guaranteed benefit to encourage traitors to buy it, since right now its usefulness is completely dependent on whether other traitors independently decided to buy it. Adding the extra channels is just a way to add that guaranteed benefit, not to bypass key theft and codewords.
> > > 
> > > 
> > > Hm, I think T encryption keys shouldn't be able to access _all_ channels, however maybe most aside from command and maybe sec?
> > 
> > 
> > Removing the command channel from the key could work, but the sec channel is probably going to be a pretty significant factor in whether or not to buy it, since knowing if sec is onto you is extremely valuable information.
> 
> Then just blast in security checkpoint or HoP office or even kill secoff.

You could do that, and risk exposing yourself. If you kill a sec officer, people are bound to notice. Blowing your way into places is also a good way to get yourself found.

**Comment by PixelTheKermit** [comment]
Having access to most channels would be great for stealth, however it's the question of what channels should be accessible. Sec, engie and maybe cargo might be good channels to make accessible? Not sure about sci, service or anything else. Would also need some slight TC cost increase as the info would be very valuable.

File: Resources/Prototypes/Entities/Objects/Devices/encryption_keys.yml [hunk 3]
```diff
@@ -274,8 +274,30 @@
 - type: entity
   parent: EncryptionKey
   id: EncryptionKeySyndie
-  name: blood-red encryption key
-  description: An encryption key used by... wait... Who is owner of this chip?
+  name: syndicate encryption key
+  description: An encryption key used by Syndicate agents to not only access the station's radio channels, but also securely contact other agents in the field.
```
Comment (AlexMorgan3817) [inline Resources/Prototypes/Entities/Objects/Devices/encryption_keys.yml:278]: Very bad desc and name. Description of abilities of it should be in uplink. This is just appearance of item, not tutorial how to ~~troll security~~ use this item

**Comment by AlexMorgan3817** [comment]
lol didn't read.
Fuck.... you still discussing this shit...
@GoodWheatley, i understand that you want to make syndie key more... usable by traitors, but still, you should look somewhere else.
Encryption key not the feature that should be changed to make people use it.
Try add some social mechanics like Super Surplus Crate, that require 2 traitors to afford it. And this will connect them and make them play together, using super bundle that they got.

**Comment by GoodWheatley** [comment]
> lol didn't read.

If you refuse to listen to what I’m saying why should I listen to you

**Comment by AlexMorgan3817** [comment]
> > lol didn't read.
> 
> If you refuse to listen to what I’m saying why should I listen to you

@GoodWheatley 
I don't force you, this is just feedback on you pull request.
You either listen or not, this is your choice.

**Comment by moonheart08** [comment]
> > > lol didn't read.
> > 
> > 
> > If you refuse to listen to what I’m saying why should I listen to you
> 
> @GoodWheatley I don't force you, this is just feedback on you pull request. You either listen or not, this is your choice.

Be polite and helpful on the github repo or you will lose the ability to contribute.

**Comment by DrSmugleaf** [comment]
> lol didn't read. Fuck.... you still discussing this shit...

Next time you antagonize another contributor like this you are getting banned from the repository and discord.

**Comment by AlexMorgan3817** [comment]
> > > > lol didn't read.
> > > 
> > > 
> > > If you refuse to listen to what I’m saying why should I listen to you
> > 
> > @GoodWheatley I don't force you, this is just feedback on you pull request. You either listen or not, this is your choice.
> 
> Be polite and helpful on the github repo or you will lose the ability to contribute.

I am sorry, guess i was very rude for a whole yesterday...

**Review by AJCM-git — CHANGES_REQUESTED** [review]
just make it worth like half a TC per channel in total (so like 6 in this case)
Though personally i think a "chamaleon" headset would be more interesting

**Comment by QuietlyWhisper** [comment]
Will this conflict with the changes made by #14415?

**Comment by Slava0135** [comment]
> Will this conflict with the changes made by #14415?

I think I should have put encryption keys stuff to different PR, as it's not essential for telecomms to be added.

**Review by AJCM-git — APPROVED** [review]

**Pull request merged**